app/containers/elements/Student/index.js

```diff
@@ -0,0 +1,206 @@
+    this.state =
+      // { ...props.student };
+      {
+        id: '',
+        firstname: '',
+        lastname: '',
+        university: '',
+        specialization: '',
+        college: '',
+        year: '',
+      };
+  }
+
+  componentDidUpdate(prevProps) {
+    const {
+      id = '',
+      firstname = '',
+      lastname = '',
+      university = '',
+      specialization = '',
+      college = '',
+      year = '',
+    } = this.props.student;
+    if (
+      prevProps.student.id !== id ||
+      prevProps.student.lastname !== lastname ||
+      prevProps.student.firstname !== firstname ||
+      prevProps.student.university !== university ||
+      prevProps.student.specialization !== specialization ||
+      prevProps.student.college !== college ||
+      prevProps.student.year !== year
+    )
+      this.setState({
+        id,
+        firstname,
+        lastname,
+        university,
+        specialization,
+        college,
+        year,
+      });
+  }
+
+  updateStudent() {
+    console.log('willUpdate', this.state.firstName, this.state.lastName);
+    this.props.updateStudent(this.state);
+    // let pdf = document.getElementById('pdfStudent').
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return this.state.id !== nextState.id ||
+      this.state.lastname !== nextState.lastname ||
+      this.state.firstname !== nextState.firstname ||
+      this.state.university !== nextState.university ||
+      this.state.college !== nextState.college ||
+      this.state.specialization !== nextState.specialization ||
+      this.state.year !== nextState.year ||
+      this.props.student.id !== nextProps.student.id ||
+      this.props.student.lastname !== nextProps.student.lastname ||
+      this.props.student.firstname !== nextProps.student.firstname ||
+      this.props.student.university !== nextProps.student.university ||
+      this.props.student.college !== nextProps.student.college ||
+      this.props.student.specialization !== nextProps.student.specialization ||
+      this.props.student.year !== nextProps.student.year;
+  }
+
+  onChange(field, event) {
+    console.log(field, event.target.value);
+    this.setState({
+      [field]: event.target.value,
+    });
+  }
+
+  render() {
+    const style = { width: 200 };
+    const {
+      id = '',
+      firstname = '',
+      lastname = '',
+      university = '',
+      specialization = '',
+      college = '',
+      year = '',
+    } = this.state;
+
+    // this.setState({ id, firstname, lastname, university, specialization, college, year });
+
+    console.log('location', this.props.location.search, this.state);
+    // console.log("in component ",id,student);
+    return (
+      <FieldsWrapper>
+        <br />
+        <TextFieldComponent
+          label="Id"
+          value={id}
+          style={style}
+          onChange={event => this.onChange('id', event)}
+        />
+        <TextFieldComponent
+          id="studentFirstName"
```

Comment:
Please add `label=First Name` as you did for the Id field, it looks good and is helpful to know what you are updating. Do the same for all the other fields.